homeassistant/core.py

```diff
@@ -99,15 +99,13 @@
 
 block_async_io.enable()
 
-T = TypeVar("T")
+_T = TypeVar("_T")
 _R = TypeVar("_R")
 _R_co = TypeVar("_R_co", covariant=True)  # pylint: disable=invalid-name
 # Internal; not helpers.typing.UNDEFINED due to circular dependency
 _UNDEF: dict[Any, Any] = {}
-# pylint: disable=invalid-name
-CALLABLE_T = TypeVar("CALLABLE_T", bound=Callable[..., Any])
-CALLBACK_TYPE = Callable[[], None]
-# pylint: enable=invalid-name
+_CallableT = TypeVar("_CallableT", bound=Callable[..., Any])
+CALLBACK_TYPE = Callable[[], None]  # pylint: disable=invalid-name
```

Comment:
Is there a reason this is in UPPER_SNAKE_CASE instead of CamelCase?
PEP 484 recommends CamelCase for Type Aliases (https://peps.python.org/pep-0484/#type-aliases)

Reply:
I would love to rename it. Unfortunately, I think it's too late for that. It's just much more common to import Type Aliases from other files. `CALLBACK_TYPE` for example is used in 84 other files and likely custom also in custom integrations. We would need a proper deprecation cycle for that.

_In contrast, I only found a very limited number of cross imports for TypeVars. Except for two integrations (`zha` and `renault`), I fixed all of them in #68154. So I believe it's safe to do rename them._